phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -4362,18 +4370,40 @@ public MutationState alterIndex(AlterIndexStatement statement) throws SQLExcepti
                     // Set so that we get the table below with the potentially modified rowKeyOrderOptimizable flag set
                     indexRef.setTable(result.getTable());
                     if (newIndexState == PIndexState.BUILDING && isAsync) {
-                        try {
-                            tableUpsert = connection.prepareStatement(UPDATE_INDEX_REBUILD_ASYNC_STATE);
-                            tableUpsert.setString(1,
-                                    connection.getTenantId() == null ? null : connection.getTenantId().getString());
-                            tableUpsert.setString(2, schemaName);
-                            tableUpsert.setString(3, indexName);
-                            tableUpsert.setLong(4, result.getTable().getTimeStamp());
-                            tableUpsert.execute();
-                            connection.commit();
-                        } finally {
-                            if (tableUpsert != null) {
-                                tableUpsert.close();
+                        if (isAll) {
+                            List<Task.TaskRecord> tasks = Task.queryTaskTable(connection, schemaName, tableName, PTable.TaskType.INDEX_REBUILD,
```

Comment:
@vincentpoon, @mihir6692 , @twdsilva , we're trying to move manually triggered index rebuilds (ALTER INDEX REBUILD) to use System.Tasks rather than the index disable timestamp + PhoenixMRJobSubmitter. Is anything else you know of depending on the syscat index timestamp being set when doing a manual rebuild?

Reply:
Are you asking if the ASYNC_REBUILD_TIMESTAMP is used for anything else? I don't think so, @vincentpoon  might know for sure.